src/libs/Navigation/AppNavigator/AuthScreens.tsx

```diff
@@ -271,6 +272,19 @@ function AuthScreens({session, lastOpenedPublicRoomID, initialLastUpdateIDApplie
             return;
         }
 
+        let signupQualifier;
```

Comment:
Adding onboarding code here disperses the onboarding logic which we want to avoid, I would move it to `src/libs/actions/Welcome/OnboardingFlow.ts` or `useOnboardingFlowRouter` for better clarity + how about adding some comments?

Reply:
I agree with @allgandalf, let's merge this PR now that's working and we can refactor the code in another PR, because it's the last PR that's holding the homepage project from closing, and an urgent feature at this point for proper conversion.